src_c/mouse.c

```diff
@@ -309,73 +325,119 @@ mouse_set_cursor(PyObject *self, PyObject *args)
 }
 
 static PyObject *
-mouse_set_system_cursor(PyObject *self, PyObject *args)
-{
-    int idnum;
+_set_system_cursor(int constant) {
+#if IS_SDLv2
     SDL_Cursor *lastcursor, *cursor = NULL;
 
-    VIDEO_INIT_CHECK();
-
-    if (!PyArg_ParseTuple(args, "i", &idnum)) {
-        return NULL;
+    cursor = SDL_CreateSystemCursor(constant);
+    if (!cursor){
+        //SDL_GetError() wasn't returning relevant stuff when this function fails
+        return RAISE(pgExc_SDLError, "Error while creating system cursor");
     }
 
+    lastcursor = SDL_GetCursor();
+    SDL_SetCursor(cursor);
+    SDL_FreeCursor(lastcursor);
+
+    cursor_data.type = 0;
+    cursor_data.constant = constant;   
+    Py_RETURN_NONE;
+#endif
+return RAISE(PyExc_TypeError, "System cursors from constant are unavailable in SDL1");
+}
+
+static PyObject *
+_set_color_cursor(int spotx, int spoty, pgSurfaceObject *surfobj) {
 #if IS_SDLv2
-    cursor = SDL_CreateSystemCursor(idnum);
-    if (!cursor) {
+    SDL_Cursor *lastcursor, *cursor = NULL;
+    SDL_Surface *surf = NULL;
+    surf = pgSurface_AsSurface(surfobj);
+
+    cursor = SDL_CreateColorCursor(surf, spotx, spoty);
+    if (!cursor)
         return RAISE(pgExc_SDLError, SDL_GetError());
-    }
+
     lastcursor = SDL_GetCursor();
     SDL_SetCursor(cursor);
     SDL_FreeCursor(lastcursor);
-#endif
+
+    cursor_data.type = 2;
+    cursor_data.spotx = spotx;
+    cursor_data.spoty = spoty;
+    cursor_data.surfobj = surfobj;
     Py_RETURN_NONE;
-}
 
+#endif
+return RAISE(PyExc_TypeError, "Cursors from a surface are unavailable in SDL1");
+}
 
-#if IS_SDLv2
 static PyObject *
-mouse_get_cursor(PyObject *self)
+mouse_set_system_cursor(PyObject *self, PyObject *args)
 {
-    return RAISE(PyExc_TypeError, "The get_cursor method is unavailable in SDL2");
+    int constant;
+    
+    VIDEO_INIT_CHECK();
+
+    if (!PyArg_ParseTuple(args, "i", &constant)) {
+        return NULL;
+    }
+
+    return _set_system_cursor(constant);
 }
-#else /* IS_SDLv1*/
+
+//this function isn't all of mouse.set_cursor, it goes through a python function first
```

Comment:
Would be good to point to where the python function lives.

Reply:
Yeah, I guess cursors.py is the best.

If we had a mouse.py, and a _mouse.c then we could combine them into one. However, this has a further complication in that we need to handle deleting old mouse.so/dll files in the setup.py.